controller/src/test/java/org/jboss/as/controller/persistence/ConfigurationFileUnitTestCase.java

```diff
@@ -0,0 +1,135 @@
+package org.jboss.as.controller.persistence;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+/**
+ *
+ * @author <a href="padamec@redhat.com">Petr Adamec</a>
+ */
+public class ConfigurationFileUnitTestCase {
+    private static final String UNCORRECT_NAME = "standalone.xml";
+    private static final String CORRECT_NAME = "20180101-000000000standalone.xml";
+    private static final String CONFIGURATION_DIR = "WFCORE-3513";
+    private static final String HISTORY_DIRECTORY = "standalone_xml_history";
+    private static final String SNAPSHOT_DIRECTORY = "snapshot";
+
+    File configurationDir;
+    File configurationFile;
+    File historyDirectory;
+    File snapshotsDirectory;
+
+    @Before
+    public void createDirectoryAndFile() throws Exception {
+        createConfigurationDir();
+        historyDirectory = new File(configurationDir, HISTORY_DIRECTORY);
+        historyDirectory.mkdir();
+        if(!historyDirectory.exists()){
+            throw new RuntimeException("cannot create dir "+HISTORY_DIRECTORY);
+        }
+        snapshotsDirectory = new File(historyDirectory, SNAPSHOT_DIRECTORY);
+        snapshotsDirectory.mkdir();
+        if(!snapshotsDirectory.exists()){
+            throw new RuntimeException("cannot create dir "+SNAPSHOT_DIRECTORY);
+        }
+    }
+
+    /**
+     * Create configuration directory for this test at tmp directory
+     */
+    private void createConfigurationDir(){
```

Comment:
Don't create files in tmp, use target instead, but this isn't really needed - see my previous comment